UI/BindingSources/CSHARP/Binding/MainViewModel.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.ObjectModel;
+using System.ComponentModel;
+
+namespace Binding;
+
+public class MainViewModel : INotifyPropertyChanged
+{
+	public event PropertyChangedEventHandler? PropertyChanged;
+
+	private ObservableCollection<Item>? _items;
+	public ObservableCollection<Item>? Items
+	{
+		get { return _items; }
+		set
+		{
+			_items = value;
+			OnPropertyChanged(nameof(Items));
+		}
+	}
+
+	public ICommand RemoveItemCommand { get; }
+
+	public MainViewModel()
+	{
+		Items = new ObservableCollection<Item>
+		{
+			new Item { Text = "Item 1" },
+			new Item { Text = "Item 2" },
+			new Item { Text = "Item 3" }
+		};
+
+		RemoveItemCommand = new RelayCommand(RemoveItem);
+	}
+
+	private void RemoveItem(object? obj)
+	{
+		if (obj is Item itemToRemove && Items is { Count: > 0})
+		{
+			Items.Remove(itemToRemove);
+		}
+	}
+
+	protected virtual void OnPropertyChanged(string propertyName)
+	{
+		PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+	}
+}
+
+public class Item
+{
+    public string? Text { get; set; }
+}
```

Comment:
```suggestion
public record Item(string? Text);
```

Reply:
Can't use it on Windows-XAML, getting the following error:

```
1>C:\UnoPlatform\Samples\Uno.Samples\UI\BindingSources\XAML\Binding\obj\Debug\net8.0-windows10.0.19041\XamlTypeInfo.g.cs(711,13,711,22): error CS8852: Init-only property or indexer 'Item.Text' can only be assigned in an object initializer, or on 'this' or 'base' in an instance constructor or an 'init' accessor.
```